src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -1,79 +1,217 @@
+        throw CallToGoogleApiFailedException(googleProject, firewallRuleName, e.getStatusCode, e.getDetails.getMessage)
+    }
+  }
+
+
+  /* Adds a firewall rule in the given google project. This firewall rule allows ingress traffic through a specified port for all
+     VMs with the network tag "leonardo". This rule should only be added once per project.
+    To think about: do we want to remove this rule if a google project no longer has any clusters? */
+  private def addFirewallRule(googleProject: GoogleProject): Future[ComputeOperation] = {
+    // Create an Allowed object that specifies the port and protocol of rule
+    //val allowedList = firewallRule.allowed.map(allowedPair => new GoogleAllowed().setIPProtocol(allowedPair.protocol).setPorts(allowedPair.port.asJava))
+    val allowed = new Allowed().setIPProtocol(proxyConfig.jupyterProtocol).setPorts(List(proxyConfig.jupyterPort.toString).asJava)
+    // Create a firewall rule that takes an Allowed object, a name, and the network tag a cluster must have to be accessed through the rule
+    val googleFirewallRule = new Firewall()
+      .setName(dataprocConfig.clusterFirewallRuleName)
+      .setTargetTags(List(dataprocConfig.clusterNetworkTag).asJava)
+      .setAllowed(List(allowed).asJava)
+
+    val request = compute.firewalls().insert(googleProject, googleFirewallRule)
+    Future {
+      executeGoogleRequest(request) // returns a ComputeOperation
+    } recoverWith {
+      case e: GoogleJsonResponseException =>
+        throw CallToGoogleApiFailedException(googleProject, dataprocConfig.clusterFirewallRuleName, e.getStatusCode, e.getDetails.getMessage)
+    }
+  }
+
+  /* Create a bucket in the given google project for the initialization files when creating a cluster */
+  def createBucket(googleProject: GoogleProject, bucketName: String): Future[Unit] = {
+    // Create lifecycle rule for the bucket that will delete the bucket after 1 day - for now, init buckets will be
+    // deleted this way, but eventually we will likely want to delete them after the cluster has been initialized
+    val lifecycleRule = new Lifecycle.Rule()
+      .setAction(new Action().setType("Delete"))
+      .setCondition(new Condition().setAge(1))
+
+    // Create lifecycle for the bucket with a list of rules
+    val lifecycle = new Lifecycle().setRule(List(lifecycleRule).asJava)
+
+    // Create a bucket object and set it's name and lifecycle
+    val bucket = new Bucket().setName(bucketName).setLifecycle(lifecycle)
+
+    val bucketInserter = storage.buckets().insert(googleProject, bucket)
+
+    Future {
+      executeGoogleRequest(bucketInserter) // returns a Bucket
+    } recoverWith {
+      case e: GoogleJsonResponseException => throw CallToGoogleApiFailedException(googleProject, bucketName, e.getStatusCode, e.getDetails.getMessage)
+    } map { bucket =>
+      BucketResponse(bucket.getName, bucket.getTimeCreated.toString)
```

Comment:
Not sure if the `BucketResponse` case class is needed since this just returns `Future[Unit]`. Maybe just `.mapTo[Unit]`?